frontend/src/components/logged_in/DeleteAcc.js

```diff
@@ -6,20 +6,29 @@ import { requestAccountDeletion } from "../../requests/user_requests";
 
 const DeleteAcc = () => {
   const { userState, setUserState } = useContext(UserContext);
+
   return (
     <div className="DeleteAcc">
       <h3>We're sad to see you go, {userState.user.username}!</h3>
       <p>If you're certain you want to delete your account forever, enter your password and press the big red button.</p>
       <input type='password' className='Password' />
       <br />
+      <br />
       <button className='btn btn-lg btn-danger' onClick={async () => {
         const pw = document.querySelector('.DeleteAcc .Password').value.trim();
-        requestAccountDeletion({ un: userState.user.username, pw: pw });
-        setUserState({
-          loggedIn: false
-        });
+        const deletion = await requestAccountDeletion({ un: userState.user.username, pw: pw });
```

Comment:
I should start doing something like this for my HTTP requests, it's way cleaner than directly calling fetch here.

Reply:
Modularity ftw